File: sync-task-scheduler/src/main/java/by/klevitov/synctaskscheduler/taskscheduler/entity/Task.java
package by.klevitov.synctaskscheduler.taskscheduler.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

import static org.apache.commons.lang3.StringUtils.isEmpty;

@Entity(name = "tasks")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = "id")
@ToString
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status;
    @Column(nullable = false)
    private String name;
    private String description;
    @Column(nullable = false)
    private String taskIdToExecute;
    @Column(nullable = false)
    private String cronExpression;

    public void copyValuesForNullOrEmptyFieldsFromTask(Task source) {
        id = (id == 0 ? source.id : id);
        status = (status == null ? source.status : status);
        name = (isEmpty(name) ? source.name : name);
        description = (isEmpty(description) ? source.getDescription() : description);
        taskIdToExecute = (isEmpty(taskIdToExecute) ? source.taskIdToExecute : taskIdToExecute);
        cronExpression = (isEmpty(cronExpression) ? source.cronExpression : cronExpression);
    }
}

File: frontend/sync-task-scheduler-ui/src/app/core/model/task.ts
import { TaskStatus } from './task-status';

export class Task {
  id: number;
  status: TaskStatus;
  name: string;
  description?: string;
  taskIdToExecute: string;
  cronExpression: string;

  constructor(
    id: number,
    status: TaskStatus,
    name: string,
    taskIdToExecute: string,
    cronExpression: string,
    description?: string
  ) {
    this.id = id;
    this.status = status;
    this.name = name;
    this.description = description;
    this.taskIdToExecute = taskIdToExecute;
    this.cronExpression = cronExpression;
  }

  static getTaskFields(): (keyof Task)[] {
    return ['status', 'name', 'description', 'taskIdToExecute', 'cronExpression'];
  }
}
